### ISSUE TITLE
LSP-SonarLint gives TypeError on on_pre_server_command

### ISSUE THREAD

**Issue by sexybiggetje** [sublimelsp/LSP-SonarLint#2]
When using LSP-SolarLint I see the follow error in the sublime console; Not sure if that breaks anything, as linting does seem to work.

`TypeError: on_pre_server_command() takes 2 positional arguments but 3 were given`

**Comment by sexybiggetje** [comment]
Sure, I don't think it has much more information close to the error, but it might. It does show another error that I have since installing this LSP.

`DPI mode: per-monitor v2
startup, version: 4126 windows x64 channel: stable
executable: /C/Program Files/Sublime Text/sublime_text.exe
application: /C/Program Files/Sublime Text
working dir: /C/Program Files/Sublime Text
packages path: /C/Users/<username>/AppData/Roaming/Sublime Text/Packages
state path: /C/Users/<username>/AppData/Roaming/Sublime Text/Local
zip path: /C/Program Files/Sublime Text/Packages
zip path: /C/Users/<username>/AppData/Roaming/Sublime Text/Installed Packages
ignored_packages: ["CSS", "TabNine", "Vintage"]
pre session restore time: 0.228915
OpenGL Context Information:
  GL API Version: 4.6.0 - Build 27.20.100.9664
  GLSL Version: 4.60 - Build 27.20.100.9664
  Vendor: Intel
  Renderer: Intel(R) UHD Graphics
startup time: 1.30592
first paint time: 1.43892
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading plugin Default.comment
reloading plugin Default.convert_color_scheme
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.html_print
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.keymap
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.ui
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin ShellScript.ShellScript
reloading plugin LESS.less_completions
reloading plugin Pretty JSON.PrettyJson
reloading plugin Pretty JSON.PrettyJsonListeners
reloading python 3.3 plugin 0_package_control_loader.00-package_control
reloading plugin Sass.sass_completions
reloading python 3.3 plugin 0_package_control_loader.01-pygments
reloading python 3.3 plugin 0_package_control_loader.01-sublime_lib
reloading python 3.3 plugin 0_package_control_loader.10-lsp_utils
reloading python 3.3 plugin 0_package_control_loader.50-backrefs
reloading python 3.3 plugin 0_package_control_loader.50-bracex
reloading python 3.3 plugin 0_package_control_loader.50-markupsafe
reloading python 3.3 plugin 0_package_control_loader.50-pathlib
reloading python 3.3 plugin 0_package_control_loader.50-pymdownx
reloading python 3.3 plugin 0_package_control_loader.50-python-markdown
reloading python 3.3 plugin 0_package_control_loader.50-pyyaml
reloading python 3.3 plugin 0_package_control_loader.51-python-jinja2
reloading python 3.3 plugin 0_package_control_loader.55-mdpopups
reloading python 3.3 plugin 0_package_control_loader.55-wcmatch
reloading python 3.3 plugin BracketHighlighter.bh_core
reloading python 3.3 plugin BracketHighlighter.bh_logging
reloading python 3.3 plugin BracketHighlighter.bh_plugin
reloading python 3.3 plugin BracketHighlighter.bh_popup
reloading python 3.3 plugin BracketHighlighter.bh_regions
reloading python 3.3 plugin BracketHighlighter.bh_remove
reloading python 3.3 plugin BracketHighlighter.bh_rules
reloading python 3.3 plugin BracketHighlighter.bh_search
reloading python 3.3 plugin BracketHighlighter.bh_swapping
reloading python 3.3 plugin BracketHighlighter.bh_wrapping
reloading python 3.3 plugin BracketHighlighter.support
reloading python 3.3 plugin CSS3.css3_completions
reloading python 3.3 plugin CSS3.rgb_to_hex
reloading python 3.3 plugin DocBlockr 2021 (DoxyDoxygen powered).Doxy
reloading python 3.3 plugin Function Name Display.FunctionNameStatus
reloading python 3.3 plugin Indent XML.indentxml
reloading python 3.3 plugin Indent XML.setup
reloading python 3.3 plugin LSP-bash.plugin
reloading python 3.3 plugin LSP-css.plugin
reloading python 3.3 plugin LSP-eslint.plugin
reloading python 3.3 plugin LSP-file-watcher-chokidar.watcher
reloading python 3.3 plugin LSP-graphql.plugin
reloading python 3.3 plugin LSP-html.plugin
reloading python 3.3 plugin LSP-intelephense.plugin
reloading python 3.3 plugin LSP-json.plugin
reloading python 3.3 plugin LSP-vue.plugin
reloading python 3.3 plugin LSP.boot
reloading python 3.3 plugin Package Control.1_reloader
reloading python 3.3 plugin Package Control.2_bootstrap
reloading python 3.3 plugin Package Control.Package Control
reloading python 3.3 plugin SqlBeautifier.sqlbeautifier
reloading python 3.3 plugin Theme - Monokai Pro.Monokai Pro
reloading python 3.3 plugin LSP-SonarLint.plugin
plugins loaded
Package Control: Installing 1 upgrades
ignored packages updated to: ["CSS", "LSP-json", "TabNine", "Vintage"]
reloading settings Packages/LSP-json/LSP-json.sublime-settings
unloading python 3.3 plugin LSP-json.plugin
reloading settings Packages/User/Package Control.sublime-settings
LSP-file-watcher-chokidar: Watcher process ended. Exception: None
reloading settings Packages/User/Preferences.sublime-settings
Package Control: Upgraded LSP-json to 1.8.1
ignored packages updated to: ["CSS", "TabNine", "Vintage"]
reloading settings Packages/LSP-json/LSP-json.sublime-settings
reloading python 3.3 plugin LSP-json.plugin
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
Unexpected exception
Traceback (most recent call last):
  File "C:\Users\<username>\AppData\Roaming\Sublime Text\Installed Packages\LSP.sublime-package\plugin/core/transports.py", line 140, in _read_loop
    payload = self._processor.read_data(self._reader)
  File "C:\Users\<username>\AppData\Roaming\Sublime Text\Installed Packages\LSP.sublime-package\plugin/core/transports.py", line 65, in read_data
    headers = http.client.parse_headers(reader)  # type: ignore
  File "./python3.3/http/client.py", line 262, in parse_headers
  File "./python3.3/socket.py", line 297, in readinto
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

Traceback (most recent call last):
  File "C:\Users\<username>\AppData\Roaming\Sublime Text\Installed Packages\LSP.sublime-package\plugin/hover.py", line 293, in run_async
  File "C:\Users\<username>\AppData\Roaming\Sublime Text\Installed Packages\LSP.sublime-package\plugin/core/sessions.py", line 1404, in run_code_action_async
  File "C:\Users\<username>\AppData\Roaming\Sublime Text\Installed Packages\LSP.sublime-package\plugin/core/sessions.py", line 1384, in execute_command
TypeError: on_pre_server_command() takes 2 positional arguments but 3 were given
font face "Times" could not be used with direct write, using gdi instead`

**Comment by rwols** [comment]
I get a different stack trace

```
Traceback (most recent call last):
  File "/Users/raoulwols/Library/Application Support/Sublime Text 3/Packages/LSP/plugin/hover.py", line 293, in run_async
    session.run_code_action_async(self._actions_by_config[config_name][index], progress=True)
  File "/Users/raoulwols/Library/Application Support/Sublime Text 3/Packages/LSP/plugin/core/sessions.py", line 1404, in run_code_action_async
    return self.execute_command(command_params, progress)
  File "/Users/raoulwols/Library/Application Support/Sublime Text 3/Packages/LSP/plugin/core/sessions.py", line 1384, in execute_command
    if self._plugin.on_pre_server_command(command, lambda: resolve(None)):
  File "/Users/raoulwols/Library/Application Support/Sublime Text 3/Packages/LSP-SonarLint/plugin.py", line 78, in on_pre_server_command
    return self._handle_open_rule_description(session, window, args, done)
  File "/Users/raoulwols/Library/Application Support/Sublime Text 3/Packages/LSP-SonarLint/plugin.py", line 101, in _handle_open_rule_description
    flags=sublime.ADD_TO_SELECTION_SEMI_TRANSIENT)
AttributeError: 'module' object has no attribute 'ADD_TO_SELECTION_SEMI_TRANSIENT'
```

@sexybiggetje are you using a modern LSP package version?

**Comment by rwols** [comment]
https://github.com/sublimelsp/LSP-SonarLint/commit/67b73715bfbd94267ab156651ebdd8b1647ab847 made it work, although the styling seems broken


https://user-images.githubusercontent.com/2431823/162572942-843d4d82-ed0a-43ae-8ab2-d575b5688b64.mov

but I believe that to be an ST core issue. Perhaps @facelessuser can chime in here because this uses MdPopups underneath.

**Comment by sexybiggetje** [comment]
Yeah I'm using the stable one from packagecontrol. This sonarlint lsp needs some love I guess? It's not in the packagecontrol repo either

**Comment by rwols** [comment]
> This sonarlint lsp needs some love I guess? It's not in the packagecontrol repo either

I'm somewhat conflicted still on how to deploy this. Download all jars when the package installed or package all jars inside of a .zip for deployment?

**Comment by sexybiggetje** [comment]
I'd say distribute them with the package, that's the most convenient if it doesn't break the license on them

**Comment by rchl** [comment]
> but I believe that to be an ST core issue

Yes, it's a known core issue. I believe it's broken in dev builds only.

**Comment by facelessuser** [comment]
> but I believe that to be an ST core issue. Perhaps @facelessuser can chime in here because this uses MdPopups underneath.

@rwols, I think I'd need a minimal reproducible example. I'm not sure what you guys are doing here. I assume you are talking about the code block backgrounds on the white page? This could be any number of things, potentially injected CSS breaking things.

**Comment by rwols** [comment]
https://github.com/sublimehq/sublime_text/issues/5343
